sktree/tests/test_supervised_forest.py

```diff
@@ -353,3 +391,185 @@ def test_check_importances_patch(criterion, dtype):
         est.fit(X, y, sample_weight=scale * sample_weight)
         importances_bis = est.feature_importances_
         assert np.abs(importances - importances_bis).mean() < tolerance
+
+
+def test_oblique_regressor_forest_orthant():
+    """Test oblique forests on orthant problem.
+
+    It is expected that axis-aligned and oblique-aligned
+    forests will perform similarly.
+    """
+    n = 500
+    X, y = _orthant(n, p=6, random_state=0)
+    n_test = 0.3
+    X_train, X_test, y_train, y_test = train_test_split(
+        X,
+        y,
+        test_size=n_test,
+        random_state=0,
+    )
+    rc_rgs = ObliqueRandomForestRegressor(max_features=None, random_state=0)
+    rc_rgs.fit(X_train, y_train)
+    y_hat = rc_rgs.predict(X_test)
+    rc_mse = mean_squared_error(y_test, y_hat)
+
+    ri_rgs = RandomForestRegressor(random_state=0)
+    ri_rgs.fit(X_train, y_train)
+    y_hat = ri_rgs.predict(X_test)
+    ri_mse = mean_squared_error(y_test, y_hat)
+
+    assert ri_mse < rc_mse
+    assert ri_mse > 0.85
+    assert rc_mse > 0.85
+
+
+def test_oblique_regressor_forest_trunk():
+    """Test oblique vs axis-aligned forests on Trunk."""
+    n = 1000
+    X, y = _trunk(n, p=100, random_state=0)
+    n_test = 0.2
+    X_train, X_test, y_train, y_test = train_test_split(
+        X,
+        y,
+        test_size=n_test,
+        random_state=0,
+    )
+
+    rc_rgs = ObliqueRandomForestRegressor(random_state=0)
+    rc_rgs.fit(X_train, y_train)
+    y_hat = rc_rgs.predict(X_test)
+    rc_mse = mean_squared_error(y_test, y_hat)
+
+    ri_rgs = RandomForestRegressor(random_state=0)
+    ri_rgs.fit(X_train, y_train)
+    y_hat = ri_rgs.predict(X_test)
+    ri_mse = mean_squared_error(y_test, y_hat)
+
+    assert ri_mse > rc_mse
+    assert ri_mse < 0.1
+    assert rc_mse < 0.1
+
+
+@pytest.mark.parametrize("dtype", (np.float64, np.float32))
+@pytest.mark.parametrize(
+    "criterion",
+    ["squared_error", "absolute_error", "friedman_mse"],
+)
+@pytest.mark.parametrize("feature_combinations", (2, 5))
+def test_check_importances_oblique_regressor(criterion, dtype, feature_combinations):
+    """Test checking feature importances for oblique trees."""
```

Comment:
You can add this test as a parametrization of the existing test ` test_check_importances_oblique`